From 5e6fe7f4266ac6ee4dc9cff4aef6bf52c8f5f689 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Tue, 21 Nov 2023 09:55:44 +0100
Subject: [PATCH 0868/1013] evl/net: socket: de-constify I/O vector for
 send/receive ops

send/receive ops may want to update the I/O vector they receive in
place, typically to perform incremental load/store from or to them,
updating iovlen on the fly. Since we allocate these vectors, pass the
corresponding handlers a mutable reference to them.

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 include/evl/net/socket.h       | 4 ++--
 kernel/evl/net/packet/packet.c | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/evl/net/socket.h b/include/evl/net/socket.h
index 334101b82319..2877fecb2e4b 100644
--- a/include/evl/net/socket.h
+++ b/include/evl/net/socket.h
@@ -36,11 +36,11 @@ struct evl_net_proto {
 		unsigned long arg);
 	ssize_t (*oob_send)(struct evl_socket *esk,
 			const struct user_oob_msghdr __user *u_msghdr,
-			const struct iovec *iov,
+			struct iovec *iov,
 			size_t iovlen);
 	ssize_t (*oob_receive)(struct evl_socket *esk,
 			struct user_oob_msghdr __user *u_msghdr,
-			const struct iovec *iov,
+			struct iovec *iov,
 			size_t iovlen);
 	__poll_t (*oob_poll)(struct evl_socket *esk,
 			struct oob_poll_wait *wait);
diff --git a/kernel/evl/net/packet/packet.c b/kernel/evl/net/packet/packet.c
index 9f25c2570765..be1422e467e5 100644
--- a/kernel/evl/net/packet/packet.c
+++ b/kernel/evl/net/packet/packet.c
@@ -408,7 +408,7 @@ static struct net_device *find_xmit_device(struct evl_socket *esk,
 /* oob */
 static ssize_t send_packet(struct evl_socket *esk,
 			const struct user_oob_msghdr __user *u_msghdr,
-			const struct iovec *iov,
+			struct iovec *iov,
 			size_t iovlen)
 {
 	struct net_device *dev, *real_dev;
@@ -563,7 +563,7 @@ static ssize_t copy_packet_to_user(struct user_oob_msghdr __user *u_msghdr,
 /* oob */
 static ssize_t receive_packet(struct evl_socket *esk,
 			struct user_oob_msghdr __user *u_msghdr,
-			const struct iovec *iov,
+			struct iovec *iov,
 			size_t iovlen)
 {
 	struct __evl_timespec uts;
-- 
2.30.2

